src/Components/SideMenu/TableList.tsx

```diff
@@ -128,53 +160,44 @@ class TableList extends React.Component<{
       partTableDict[partTableNameSplitResult[0]].push(new PartTableListEntry(partTableNameSplitResult[1]));
     }
 
-    // Parse through the rest of the table types of Computed, Manual, Imported and Lookup and attach Part table accordingly. Ignore all other type
-    let tableListDictKeys: Array<string> = Object.keys(this.props.tableListDict);
-
     // Create a new tableList to later use for setState
     let tableList: Array<ParentTableListEntry> = [];
 
-    // Remove part_tables entry from the key list
-    tableListDictKeys.splice(tableListDictKeys.indexOf('part_tables'));
+    this.parseTableEntry(tableList, this.props.tableListDict.computed_tables, TableType.COMPUTED, partTableDict);
+    this.parseTableEntry(tableList, this.props.tableListDict.manual_tables, TableType.MANUAL, partTableDict);
+    this.parseTableEntry(tableList, this.props.tableListDict.lookup_tables, TableType.LOOKUP, partTableDict);
+    this.parseTableEntry(tableList, this.props.tableListDict.imported_tables, TableType.IMPORTED, partTableDict);
 
-    // Looped through each type of table that is not part
-    for (let tableTypeName of tableListDictKeys) {
-      // Figure out what table type to be set
-      let tableType = null;
+     // Update the state and reset sort mode to ATOZ
+     this.setState({tableList: tableList, restrictedTableList: tableList, searchString: '', currentTableSortMode: TableSortMode.ATOZ});
+  }
 
-      if (tableTypeName === 'computed_tables') {
-        tableType = TableType.COMPUTED;
-      }
-      else if (tableTypeName === 'manual_tables') {
-        tableType = TableType.MANUAL;
-      }
-      else if (tableTypeName === 'lookup_tables') {
-        tableType = TableType.LOOKUP;
-      }
-      else if (tableTypeName === 'imported_tables') {
-        tableType = TableType.IMPORTED;
+  /**
+   * Given a list of parent table names, check if there is any partTables that belong to it, if so attach it then push to tableList, otherwise just push
+   * @param tableList 
+   * @param tableNames 
+   * @param tableType 
+   * @param partTableDict 
+   */
+  parseTableEntry(tableList: Array<ParentTableListEntry>, tableNames: Array<string>, tableType: TableType, partTableDict: Record<string, Array<PartTableListEntry>>) {
+    console.log(tableNames)
```

Comment:
```suggestion
```
i spy a console.log

Reply:
ah thanks, was trying to figure out why my quick rewrite broke the code